drivers/hid/i2c-hid/i2c-hid-core.c

```diff
@@ -182,8 +183,10 @@ static const struct i2c_hid_quirks {
 		I2C_HID_QUIRK_NO_RUNTIME_PM },
 	{ I2C_VENDOR_ID_GOODIX, I2C_DEVICE_ID_GOODIX_01F0,
 		I2C_HID_QUIRK_NO_RUNTIME_PM },
-	{ USB_VENDOR_ID_ELAN, I2C_PRODUCT_ID_ELAN_TOUCHPAD,
-		I2C_HID_QUIRK_FORCE_TRIGGER_FALLING },
+	{ USB_VENDOR_ID_ELAN, HID_ANY_ID,
+		I2C_HID_QUIRK_BOGUS_IRQ },
+	{ USB_VENDOR_ID_ELAN, USB_DEVICE_ID_ELAN_TOUCHPAD,
+		I2C_HID_QUIRK_BOGUS_IRQ | I2C_HID_QUIRK_FORCE_TRIGGER_FALLING },
```

Comment:
As I tried to explain before, the { USB_VENDOR_ID_ELAN, USB_DEVICE_ID_ELAN_TOUCHPAD,  ...} quirk MUST be above the { USB_VENDOR_ID_ELAN, HID_ANY_ID, ...} quirk. The quirks are checked one by one until one matches.

Except that I'm wrong and i2c_hid_lookup_quirk keeps going after a match, so the most specific quirk must be at the bottom.

Never mind this is correct, lets discuss this further on the mailinglist.

Reply:
You are right because i2c_hid_lookup_quirk looks from the top.